tds/modules/model/utils.py

```diff
@@ -51,9 +90,13 @@ def model_response(model_from_es, delete_fields=None) -> dict:
     # model["state_id"] = es_response["_id"]
     # frameworks = get_frameworks()
     # model["framework"] = frameworks.get(model["model_schema"], model["model_schema"])
-    if "model_schema" in model:
-        model["schema"] = model["model_schema"]
-        del model["model_schema"]
+
+    model = restructure_model_header(model)
+
+    # Rename 'schema' to 'model_schema' if present
+    if "model_schema" in model["header"]:
```

Comment:
This check could probably be folded in to the `restructure_model_header()` call if that makes sense. I'll put a code suggestion in a different comment.